### PR TITLE
formatter: Add RESPONSE_FLAGS_FULL substitution formatter

### LINKED ISSUE #28572: Full text for RESPONSE_FLAGS in access logs
The ``%RESPONSE_FLAGS%`` access log flag will output additional information in values which may not be understood (most of them) unless [referring to the docs](https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log/usage). This feature request is for additional flag ``%RESPONSE_FLAGS_FULL%`` that would support the same flags, but will full text description in pascal case. For example, ``UH`` would map to ``UnhealthyUpstream``.

### PR DESCRIPTION
**Pull request by ohadvano**
Additional Description: Adds ``RESPONSE_FLAGS`` equivalent which outputs PascalCase strings for readability purposes. Resolves #28572
Risk Level: Low
Testing: Unit tests
Docs Changes: Changelogs, access log
Release Notes: None
Platform Specific Features: None

### CODE REVIEW

File: source/common/stream_info/utility.h
Comment (ggreenway) [inline source/common/stream_info/utility.h:119]: Instead of a separate array, can you make this a single array (combined with above) with type std::pair (or struct with better field names) for the short and long name? That will prevent errors when adding a flag if someone only adds to one array and not the other.

**Review by ggreenway — CHANGES_REQUESTED** [review]
/wait

**Comment by ggreenway** [comment]
@ohadvano please leave it to the maintainers to assign PRs. Among other reasons, you got a 5 day delay because I was on vacation.

File: source/common/stream_info/utility.h
Comment (ohadvano) [reply source/common/stream_info/utility.h]: Good callout, thanks

**Comment by ohadvano** [comment]
@ggreenway I adjusted per your comment, can you please take another look?

File: docs/root/configuration/observability/access_log/usage.rst
Comment (ggreenway) [inline docs/root/configuration/observability/access_log/usage.rst:494]: I think it would be better to combine the docs for RESPONSE_FLAGS_FULL and RESPONSE_FLAGS to prevent drift. Maybe make the RESPONSE_FLAGS section mention both formatters, and on each line list both the short string and full string, followed by the docs for that line.

File: docs/root/configuration/observability/access_log/usage.rst
Comment (ggreenway) [inline docs/root/configuration/observability/access_log/usage.rst:489]: I think I'd prefer renaming this to `RESPONSE_FLAGS_LONG`

**Review by ggreenway — CHANGES_REQUESTED** [review]
The code looks good for this; just a few notes on docs and naming

/wait

File: docs/root/configuration/observability/access_log/usage.rst
Comment (ohadvano) [reply docs/root/configuration/observability/access_log/usage.rst]: done

**Review by ggreenway — APPROVED** [review]
LGTM

**Pull request merged**
